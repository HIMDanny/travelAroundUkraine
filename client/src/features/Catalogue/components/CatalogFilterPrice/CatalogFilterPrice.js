import React, { useState } from 'react';
import { styled, alpha, Box, Slider, InputBase } from '@mui/material';
import { FilterAccordion } from '..';

const minPrice = 10;
const maxPrice = 1000;

function valuetext(value) {
  return `${value} â‚¬`;
}

const FilterInput = styled(InputBase)(({ theme }) => ({
  '& .MuiInputBase-input': {
    borderRadius: 40,
    border: '1px solid #EDEDED',
    width: 105,
    height: 30,
    paddingLeft: 16,
    fontSize: 12,
    boxSizing: 'border-box',

    '&:focus': {
      boxShadow: `${alpha(theme.palette.primary.main, 0.15)} 0 0 0 0.12rem`,
      borderColor: theme.palette.primary.main,
    },
  },
}));

const CatalogFilterPrice = () => {
  const [value, setValue] = useState([200, 600]);

  const handleChange = (event, newValue) => {
    setValue(newValue);
  };

  return (
    <FilterAccordion title="Price">
      <Box sx={{ display: 'flex', justifyContent: 'space-between' }}>
        <FilterInput placeholder={String(minPrice)} value={value[0]} />
        <FilterInput placeholder={String(maxPrice)} value={value[1]} />
      </Box>
      <Box sx={{ m: '0 auto', pt: '20px' }}>
        <Slider
          min={minPrice}
          max={maxPrice}
          getAriaLabel={() => 'Price range'}
          value={value}
          onChange={handleChange}
          getAriaValueText={valuetext}
        />
      </Box>
    </FilterAccordion>
  );
};

export default CatalogFilterPrice;
