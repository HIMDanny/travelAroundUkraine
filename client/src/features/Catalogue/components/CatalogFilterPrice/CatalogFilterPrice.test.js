import { render } from '@testing-library/react';
import CatalogFilterPrice from './CatalogFilterPrice';

jest.mock('@mui/material/Box', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Slider', () => ({ children }) => <span>{children}</span>);
jest.mock('@mui/material/InputBase', () => ({ children }) => <input>{children}</input>);


describe('CatalogFilterPrice snapshot test', () => {
  test('should CatalogFilterPrice match snapshot', () => {
    const { asFragment } = render(<CatalogFilterPrice />);
    expect(asFragment()).toMatchSnapshot();
  });
});
