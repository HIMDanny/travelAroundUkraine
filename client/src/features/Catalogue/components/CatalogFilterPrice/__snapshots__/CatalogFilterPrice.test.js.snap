// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogFilterPrice snapshot test should CatalogFilterPrice match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root Mui-expanded MuiAccordion-gutters css-18xlkcq-MuiPaper-root-MuiAccordion-root"
  >
    <div
      aria-expanded="true"
      class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded MuiAccordionSummary-gutters css-1t26u27-MuiButtonBase-root-MuiAccordionSummary-root"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content Mui-expanded MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-16as4g4-MuiTypography-root"
        >
          Price
        </p>
      </div>
      <div
        class="MuiAccordionSummary-expandIconWrapper Mui-expanded css-yw020d-MuiAccordionSummary-expandIconWrapper"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            class="MuiAccordion-region"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root css-1cwf0m9-MuiAccordionDetails-root"
            >
              <div>
                <input />
                <input />
              </div>
              <div>
                <span />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
